TypeSystem/Serialization/json.h

```diff
@@ -195,17 +195,32 @@ struct SaveIntoJSONImpl {
     }
   };
 
+  // No-op function required for compilation.
+  template <typename TT = T>
+  static ENABLE_IF<std::is_same<TT, CurrentSuper>::value> Save(rapidjson::Value&,
+                                                               rapidjson::Document::AllocatorType&,
+                                                               const TT&,
+                                                               bool) {}
   // `CURRENT_STRUCT`.
   template <typename TT = T>
-  static ENABLE_IF<IS_CURRENT_STRUCT(TT)> Save(rapidjson::Value& destination,
-                                               rapidjson::Document::AllocatorType& allocator,
-                                               const TT& source) {
-    destination.SetObject();
+  static ENABLE_IF<IS_CURRENT_STRUCT(TT) && !std::is_same<TT, CurrentSuper>::value> Save(
+      rapidjson::Value& destination,
+      rapidjson::Document::AllocatorType& allocator,
+      const TT& source,
+      bool save_as_super = false) {
```

Comment:
The name doesn't reflect what it does. Some `bool set_object_already_called = false`?